src/cache.c

```diff
@@ -35,20 +35,66 @@ static pthread_mutex_t cf_lock;
  */
 static char *DATA_DIR;
 
-static char *CacheSystem_get_cache_home()
+
+/**
+ * \brief Return the fullpath to the cache or config directory
+ *
+ * \param[in] cache_or_config 1 to get cache home, 0 for config
+ */
+static char *CacheSystem_get_home(int cache_or_config)
```

Comment:
I think it is better to split this function into two. I don't mind there being a bit of code duplication. The point is that the variable names here are not very clear, and the purpose of this function is just a bit confusing right now. 

I would have ``CacheSystem_get_cache_dir()`` and a ``get_XDG_CONFIG_HOME``. This is because ``CacheSystem_get_config_home`` is only used in ``main.c``, and you are using the output to get the actualy location of ``httpdirfs/config``, it goes beyond the remit of the cache system. You could leave it as a static function in ``main.c``, or put it into ``util.c`` if you think it might get used elsewhere.